modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationMetricsListener.java

```diff
@@ -82,10 +83,22 @@
     private final Function<Short, LongAdderMetric> rcvdMsgsCntByTypeMetricFactory;
 
     /** Function to be used in {@link Map#computeIfAbsent(Object, Function)} of {@code sentMsgsMetricsByNodeId}. */
-    private final Function<UUID, LongAdderMetric> sentMsgsCntByNodeIdMetricFactory;
+    private final Function<UUID, AtomicLong> sentMsgsCntByNodeIdMetricFactory;
```

Comment:
Why this change?

Reply:
Are we still need this factory?

Reply:
These things won't be exported as metrics. We have them for JMX support only.

Reply:
And yes, factory saves us from extra allocations, we discussed this privately.

Reply:
Can we map `consistentId -> nodeId` when the user requests data to reduce the amount of metrics we store internally?

Reply:
```
        Map<UUID, Long> msgs = metricsLsnr.receivedMessagesByNode();

        Map<Object, Long> consistentIdMap = new HashMap<>();

        IgniteCluster cluster = ignite().cluster();

        for (Map.Entry<UUID, Long> entry : msgs.entrySet()) 
            consistentIdMap.put(cluster.node(entry.getKey()).consistentId(), entry.getValue());
        
        return consistentIdMap;
```